src/babashka/nrepl/impl/server.clj

```diff
@@ -162,14 +163,53 @@
                      "type" (cond
                               (ifn? (:val m)) "function"
                               :else "variable")
-                     "docstring" (:doc m)
+                     "docstring" (:doc m "")
                      "status" #{"done"}}]
           (utils/send os
                       (utils/response-for msg reply) opts)))
       (catch Throwable e
         (when debug (println e))
         (utils/send os (utils/response-for msg {"status" #{"done" "no-eldoc"}}) opts)))))
 
+(defn lookup [ctx os msg {:keys [debug] :as opts}]
+  (try
+    (let [ns-str (:ns msg)
+          sym-str (:symbol msg)
+          sym-ns (some-> (symbol sym-str)
+                         namespace
+                         symbol)
+          curr-ns-aliases (sci/eval-string*
+                           ctx
+                           (format "(ns-aliases '%s)" @vars/current-ns))
+          sym-aliased? (contains? (set (keys curr-ns-aliases)) sym-ns)
+          unaliased-sym-ns-str (when sym-aliased?
+                                 (str (get curr-ns-aliases sym-ns)))
+          sci-ns (when ns-str
+                   (sci-utils/namespace-object (:env ctx) (symbol ns-str) nil false))]
+      (sci/binding [vars/current-ns (or sci-ns @vars/current-ns)]
+        (let [m (sci/eval-string*
+                 ctx
+                 (format "
+(when-let [v (ns-resolve '%s '%s)]
+  (let [m (meta v)]
+    (assoc m :arglists (:arglists m)     
+     :doc (:doc m)
+     :name (:name m)
+     :ns (some-> m :ns ns-name)
+     )))"
+                         (if sym-aliased? unaliased-sym-ns-str ns-str)
+                         (if sym-aliased? (utils/strip-symbol-str-ns sym-str) sym-str)))
```

Comment:
I suggest just calling `name` on  the symbol created on line 178 (move it to a let binding) and removing this util function.